src/poll.rs

```diff
@@ -132,6 +135,255 @@ libc_bitflags! {
     }
 }
 
+/// Timeout argument for [`poll`].
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct PollTimeout(i32);
+
+/// Error type for [`PollTimeout::try_from::<i128>::()`].
+#[derive(Debug, Clone, Copy)]
+pub enum TryFromI128Error {
+    /// Value is less than -1.
+    Underflow(crate::Errno),
+    /// Value is greater than [`i32::MAX`].
+    Overflow(<i32 as TryFrom<i128>>::Error),
+}
+impl fmt::Display for TryFromI128Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Underflow(err) => write!(f, "Underflow: {err}"),
+            Self::Overflow(err) => write!(f, "Overflow: {err}"),
+        }
+    }
+}
+impl std::error::Error for TryFromI128Error {}
+
+/// Error type for [`PollTimeout::try_from::<i68>()`].
+#[derive(Debug, Clone, Copy)]
+pub enum TryFromI64Error {
+    /// Value is less than -1.
+    Underflow(crate::Errno),
+    /// Value is greater than [`i32::MAX`].
+    Overflow(<i32 as TryFrom<i64>>::Error),
+}
+impl fmt::Display for TryFromI64Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Underflow(err) => write!(f, "Underflow: {err}"),
+            Self::Overflow(err) => write!(f, "Overflow: {err}"),
+        }
+    }
+}
+impl std::error::Error for TryFromI64Error {}
+
+// These cases implement slightly different conversions that make using generics impossible without
+// specialization.
+impl PollTimeout {
+    /// Blocks indefinitely.
+    pub const NONE: Self = Self(-1);
+    /// Returns immediately.
+    pub const ZERO: Self = Self(0);
+    /// Blocks for at most [`std::i32::MAX`] milliseconds.
+    pub const MAX: Self = Self(i32::MAX);
+    /// Returns if `self` equals [`PollTimeout::NONE`].
+    pub fn is_none(&self) -> bool {
+        *self == Self::NONE
```

Comment:
Just to program defensively, I recommend:
```suggestion
        *self <= Self::NONE
```

Reply:
You've right, fixed it.